src/emcee/moves/stretch.py

```diff
@@ -24,10 +24,18 @@ def __init__(self, a=2.0, **kwargs):
         super(StretchMove, self).__init__(**kwargs)
 
     def get_proposal(self, s, c, random):
+        try:
+            rg_integers = random.integers
+        except AttributeError:
+            rg_integers = random.randint
         c = np.concatenate(c, axis=0)
         Ns, Nc = len(s), len(c)
         ndim = s.shape[1]
-        zz = ((self.a - 1.0) * random.rand(Ns) + 1) ** 2.0 / self.a
+        try:
```

Comment:
Instead of this try/except block you could use `random.uniform`.

```
zz = ((self.a - 1.0) * random.uniform(size=Ns) + 1) ** 2.0 / self.a
```
`uniform` also accepts the `low` and `high` keywords, so you might be able to simplify the term in the parenthesis.

Reply:
I think it simplifies to (needs checking):
```
zz = (random.uniform(low=1, high=self.a, size=Ns)) ** 2.0 / self.a
```